### PR TITLE
Added retry checking into do_get in the api.  A 'retry' (429) will be…

### PR DESCRIPTION
**Pull request by zpeters**
… returned as an 'ok' status. We will sleep for the specified time and retry thequery transparently

### CODE REVIEW

File: lib/ex_pwned/api.ex [hunk 1]
```diff
@@ -18,10 +18,23 @@ defmodule ExPwned.Api do
         do: do_get(path_arg, Enum.into(query_params, %{}))
 
       def do_get(path_arg, query_params) do
-        path_arg
-        |> build_url(query_params)
-        |> HTTPoison.get(headers())
-        |> Parser.parse()
+        {:ok, result, status} =
```
Comment (techgaun) [inline lib/ex_pwned/api.ex:21]: seems like this matching will not work for error case that return error tuple.

**Review by techgaun — CHANGES_REQUESTED** [review]
thanks @zpeters very useful change but this will now raise for error tuples. could you add a case for error tuples handling.

File: lib/ex_pwned/api.ex (hunk 1 above)
Comment (zpeters) [reply lib/ex_pwned/api.ex:21]: Question for you.  I have asked a few people online and their recommendation is to not retry in the library, but to error and emit the retry timeout.  Letting the application using the library to determine how to handle the response.  Does this sound like a better approach to you?

Comment (techgaun) [reply lib/ex_pwned/api.ex:21]: That is definitely something I agree with. Just need to emit the timeout error information that includes timeout value before another retry.

Comment (zpeters) [reply lib/ex_pwned/api.ex:21]: Let's do this.  I will send a different PR for the new approach

**Comment by zpeters** [comment]
Withdrawn

**Pull request closed without merging**